types/es-abstract/helpers/isSamePropertyDescriptor.d.ts

```diff
@@ -1,7 +1,11 @@
 import { PropertyDescriptor as ESPropertyDescriptor } from '../index';
 
 declare function isSamePropertyDescriptor(
-    ES: { SameValue(x: unknown, y: unknown): boolean },
+    ES: {
+        // This is necessary to avoid "TS2345: Object literal may only specify known properties":
+        [method: string]: any;
+        SameValue(x: unknown, y: unknown): boolean;
```

Comment:
```suggestion
        SameValue(x: unknown, y: unknown): x is typeof y;
```

Reply:
That just result in `x is unknown`

Reply:
oh right, hm. what about:
```suggestion
        SameValue<T = unknown, S = unknown>(x: T, y: S): x is S;
```
?

Reply:
would something like my suggestion work?

Reply:
That would make the es‑abstract module unassignable to the `ES` parameter.

Reply:
Can you help me understand why? `es-abstract`'s export should be a type that extends the base ES type..

Reply:
`es‑abstract`’s export extends `es‑abstract/es2015`.

This is about the type of the parameter named `ES` in `es‑abstract/helpers/*.js`.

Reply:
sure - but that parameter is always a `Partial` of the union of "all ES year types"